Common/Global/RecipeOverride.cs

```diff
@@ -0,0 +1,35 @@
+﻿using CalamityMod.Items.Materials;
+using CalamityMod.Tiles.Furniture.CraftingStations;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Terraria;
+using Terraria.ModLoader;
+using TheBereftSouls.Common.Utility;
+
+namespace TheBereftSouls.Common.Global
+{
+    public class RecipeOverride : ModSystem
+    {
+
+        public override void PostAddRecipes()
+        {
+
+            foreach (Recipe recipe in Main.recipe)
+            {
+                if (TheBereftSouls.CalamityRangerExpansion != null)
+                {
+                    if (recipe.TryGetResult(ExternalModCallUtils.GetItemFromMod(TheBereftSouls.CalamityRangerExpansion, "AutoCalculationCoil"), out _))
+                    {
+                        recipe.AddIngredient<CosmiliteBar>(5);
+                        RecipeUtils.OverrideTile<CosmicAnvil>(recipe);
+                        recipe.AddTile<CosmicAnvil>();
+                    }
+                }
```

Comment:
Could be simplified to the following:

```csharp
                if (TheBereftSouls.CalamityRangerExpansion == null)
                {
                    continue;
                }
                if (!recipe.TryGetResult(ExternalModCallUtils.GetItemFromMod(TheBereftSouls.CalamityRangerExpansion, "AutoCalculationCoil"), out _))
                {
                    continue;
                }
                recipe.AddIngredient<CosmiliteBar>(5);
                RecipeUtils.OverrideTile<CosmicAnvil>(recipe);
                recipe.AddTile<CosmicAnvil>();
```

Reason: Less indentation and easier to follow through.

> [!NOTE]
> This was also mentioned in #9 in this comment: https://github.com/bromeex/TheBereftSouls/pull/9#issuecomment-2387960305

Reply:
The idea with this indentation is to be able to stake several changes within that, I only left one for the moment since I didn't know what other changes to add, it seems to me that doing it this way simplifies understanding due to the indentation

Reply:
That way you propose, I think it has the problem of not being scalable.